esp32s3-hal/Cargo.toml

```diff
@@ -29,6 +29,7 @@ embedded-hal   = { version = "0.2",  features = ["unproven"] }
 embedded-hal-1 = { package = "embedded-hal", version = "=1.0.0-alpha.8" }
 xtensa-lx      = { version = "0.7",  features = ["esp32s3"] }
 xtensa-lx-rt   = { version = "0.12", features = ["esp32s3"], optional = true }
+r0             = { version = "1.0.0", optional = true }
```

Comment:
Haven't looked at the PR fully yet, but it seems that r0 is deprecated: https://github.com/rust-embedded/r0, should we be using it?

Reply:
`r0` is already a dependency of `esp32c3-hal` so I'm not totally again pulling it in, but thanks for bringing that up. At the very least we should open an issue for exorcising `r0`.

Reply:
Also xtensa-lx-rt depends on it afaik - so I used it for now until we get rid of it all together. Don't know what riscv-rt is planning or have done about it

Reply:
Ah didn't know riscv was still using it. I don't think it's too bad to use, from what I can tell the cortex-m reason for switching was _potential_ UB in the future. Not sure if that's ever been exercised in reality though...

Let's keep using it for now, and try and find an alternative in the future.